File: backend/app/models/driver_document.py
import uuid
from datetime import date

from sqlalchemy import Date, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base
from app.models.base import TimestampMixin, UUIDPKMixin


class DriverDocument(Base, UUIDPKMixin, TimestampMixin):
    __tablename__ = "driver_documents"

    company_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("companies.id", ondelete="CASCADE"), nullable=False
    )
    driver_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("drivers.id", ondelete="CASCADE"), nullable=False
    )
    document_type_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("driver_document_types.id"), nullable=False
    )
    number: Mapped[str | None] = mapped_column(String(100), nullable=True)
    expiry_date: Mapped[date | None] = mapped_column(Date, nullable=True)
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)

    document_type: Mapped["DriverDocumentType"] = relationship()

File: frontend/src/types/driver_document.ts
export interface DriverDocumentType {
  id: string
  company_id: string
  name: string
  alert_days_before: number
  created_at: string
  updated_at: string
}

export interface DriverDocument {
  id: string
  company_id: string
  driver_id: string
  document_type_id: string
  number: string | null
  expiry_date: string | null
  notes: string | null
  created_at: string
  updated_at: string
  document_type: DriverDocumentType
}
